Specify java version

```
2021-06-24 01:39:39,488 [Thread-1] INFO : [INFO] ------------------------------------------------------------------------
2021-06-24 01:39:39,488 [Thread-1] INFO : [INFO] BUILD FAILURE
2021-06-24 01:39:39,488 [Thread-1] INFO : [INFO] ------------------------------------------------------------------------
2021-06-24 01:39:39,488 [Thread-1] INFO : [INFO] Total time:  9.811 s
2021-06-24 01:39:39,489 [Thread-1] INFO : [INFO] Finished at: 2021-06-24T01:39:39+05:30
2021-06-24 01:39:39,489 [Thread-1] INFO : [INFO] ------------------------------------------------------------------------
2021-06-24 01:39:39,491 [Thread-1] INFO : [ERROR] Failed to execute goal org.apache.maven.plugins:maven-plugin-plugin:3.6.1:descriptor (default-descriptor) on project hadoop-maven-plugins: The API of the mojo scanner is not compatible with this plugin version. Please check the plugin dependencies configured in the POM and ensure the versions match. Can't load library: /nix/store/qybd7j6v7kb7yhizc7gklgg3lyrxf38y-openjdk-headless-11.0.8+10/lib/openjdk/lib/libawt_xawt.so -> [Help 1]
```

Running into this issue when I try to use mvn2nix on hadoop 3.3.1. This probably happens because hadoop uses Java 8. Is it possible to specify the Java version used by mvn2nix?

--- a/src/main/java/com/fzakaria/mvn2nix/cmd/Maven2nix.java
+++ b/src/main/java/com/fzakaria/mvn2nix/cmd/Maven2nix.java
@@ -57,6 +57,12 @@ public class Maven2nix implements Callable<Integer> {
             defaultValue = "https://repo.maven.apache.org/maven2/")
     private String[] repositories;
 
+    @Option(names = "--jdk",
+            arity = "0..*",
+            description = "The JDK to use when running Maven",
+            defaultValue = "${java.home}")
+    private File javaHome;
+    
     public Maven2nix() {
     }
 
@@ -65,7 +71,7 @@ public class Maven2nix implements Callable<Integer> {
         LOGGER.info("Reading {}", file);
 
         final Maven maven = Maven.withTemporaryLocalRepository();
-        maven.executeGoals(file, goals);
+        maven.executeGoals(file, javaHome, goals);
 
         Collection<Artifact> artifacts = maven.collectAllArtifactsInLocalRepository();
         Map<String, MavenArtifact> dependencies = artifacts.parallelStream()

--- a/src/main/java/com/fzakaria/mvn2nix/maven/Maven.java
+++ b/src/main/java/com/fzakaria/mvn2nix/maven/Maven.java
@@ -89,11 +89,12 @@ public class Maven {
         return Optional.of(file);
     }
 
-    public void executeGoals(File pom, String... goals) throws MavenInvocationException {
+    public void executeGoals(File pom, File javaHome, String... goals) throws MavenInvocationException {
         InvocationRequest request = new DefaultInvocationRequest();
         request.setGoals(Lists.newArrayList(goals));
         request.setBatchMode(true);
         request.setPomFile(pom);
+        request.setJavaHome(javaHome);
 
         /*
          * Load a custom settings.xml file that sets ~/.m2/repository as a remote repo.

